src/BusinessLogic/ShipmentDraft/ShipmentDraftService.php

```diff
@@ -139,4 +153,86 @@ protected function enqueueDelayedTask(SendDraftTask $task, $delayInterval)
 
         RepositoryRegistry::getRepository(Schedule::CLASS_NAME)->save($schedule);
     }
+
+    /**
+     * Enqueues shipment related schedules.
+     *
+     * @throws \Logeecom\Infrastructure\ORM\Exceptions\RepositoryNotRegisteredException
+     */
+    private function enqueueShipmentSchedules()
+    {
+        $repository = RepositoryRegistry::getRepository(Schedule::getClassName());
+        $firstStart = rand(0, 59);
+        $secondStart = ($firstStart + 30) % 60;
+
+        // Schedule hourly task for updating shipment info - start at full hour
+        $this->scheduleUpdatePendingShipmentsData($repository, $firstStart);
+
+        // Schedule hourly task for updating shipment info - start at half hour
+        $this->scheduleUpdatePendingShipmentsData($repository, $secondStart);
+
+        // Schedule daily task for updating shipment info - start at 11:00 UTC hour
+        $this->scheduleUpdateInProgressShipments($repository, 11, rand(0, 59));
```

Comment:
Time between firstStart and secondStart,  and start time(11:00) should be in db or configuration-file... Something more easy to change in case of over load. Any change in this simple values will end up with PR+Review+Release flow.

Reply:
The times are explicitly requested. The first two are executed every hour each, so we need exactly 30 minutes between them - most optimal for any user and system. If this needs to change, then we need to remove one hourly schedule, and this should be another PR.
For the last one, it is explicitly requested to be at 11:00 UTC. We can add a configuration for this as well, but it would only change the time, not randomize times for all users. To change it to be random or similar, it would still need another PR.

Reply:
👍